src/Cli/dotnet/commands/dotnet-restore/RestoreCommandParser.cs

```diff
@@ -25,6 +25,13 @@ internal static class RestoreCommandParser
         }.ForwardAsSingle(o => $"-property:RestoreSources={string.Join("%3B", o)}")
         .AllowSingleArgPerToken();
 
+        public static readonly CliOption<string> FrameworkOption = new ForwardedOption<string>("--framework")
+        {
+            Description = LocalizableStrings.CmdFrameworkOptionDescription,
+            HelpName = LocalizableStrings.CmdFrameworkArgument
+        }.ForwardAsSingle(o => $"-property:TargetFramework={o}")
+        .AddCompletions(Complete.TargetFrameworksFromProjectFile);
+
```

Comment:
```suggestion
        public static readonly CliOption<IEnumerable<string>> FrameworkOption = new ForwardedOption<IEnumerable<string>>("--framework")
        {
            Description = LocalizableStrings.CmdFrameworkOptionDescription,
            HelpName = LocalizableStrings.CmdFrameworkArgument
        }.ForwardAsSingle(o => $"-property:TargetFrameworks={string.Join(";", o)}")
        .AllowSingleArgPerToken()
        .AddCompletions(Complete.TargetFrameworksFromProjectFile);
```

We can probably change the option this way in case we would like to allow multiple `--framework` option to be specified.